### PR TITLE
Fix some NPE and upgrade dependencies

### PR DESCRIPTION
**Pull request by devnied**
This PR fix some issues found on some devices:

- `currentValue` used instead of `value` and produce NPE
- `assert bundle != null;` is never reach in a release app, so we need to check is the bundle is not null.
- Expression from parcel read list of operators in the wrong variable
- Add conditional check due to issue on some devices

Bump appcompat dependencies to version 1.1.0

### CODE REVIEW

**Review by maltaisn — COMMENTED** [review]
Thanks for the fixes! 

When is `nbFormat` null? If null value is caused by deserialization, wouldn't it be better to set a fallback value immediately so that the field is never null? Same thing for `expression` in the presenter.

File: calcdialog/src/main/java/com/maltaisn/calcdialog/CalcPresenter.java [hunk 1]
```diff
@@ -113,7 +114,9 @@ void detach() {
     }
 
     void writeStateToBundle(Bundle bundle) {
-        bundle.putParcelable("expression", expression);
+        if (expression != null) {
+            bundle.putParcelable("expression", expression);
+        }
```
Comment (maltaisn) [inline calcdialog/src/main/java/com/maltaisn/calcdialog/CalcPresenter.java]: How can the expression be null?

**Comment by devnied** [comment]
Yes on some devices the deserialization put null on `nbFormat` and/or `expression`.
I agree for the default value, I will fix the PR

**Comment by maltaisn** [comment]
I already made the changes locally, I'll make a new release soon. Thanks again!

**Pull request merged**